Extensions/Terraform/Src/Tasks/TerraformInstaller/src/terraform-installer.ts

```diff
@@ -0,0 +1,92 @@
+import tasks = require('azure-pipelines-task-lib/task');
+import tools = require('azure-pipelines-tool-lib/tool');
+import path = require('path');
+import os = require('os');
+import fs = require('fs');
+
+const uuidV4 = require('uuid/v4');
+const terraformToolName = "terraform";
+const isWindows = os.type().match(/^Win/);
+
+export async function downloadTerraform(inputVersion: string): Promise<string> {
+    let version = tools.cleanVersion(inputVersion);
+    if (!version) {
+        throw new Error(tasks.loc("InputVersionNotValidSemanticVersion", inputVersion));
+    }
+
+    let cachedToolPath = tools.findLocalTool(terraformToolName, version);
+    if (!cachedToolPath) {
+        let terraformDownloadUrl = getTerraformDownloadUrl(version);
+        let fileName = `${terraformToolName}-${version}-${uuidV4()}.zip`;
+        let terraformDownloadPath;
+
+        try {
+            terraformDownloadPath = await tools.downloadTool(terraformDownloadUrl, fileName);
+        } catch (exception) {
+            throw new Error(tasks.loc("TerraformDownloadFailed", terraformDownloadUrl, exception));
+        }
+
+        let terraformUnzippedPath = await tools.extractZip(terraformDownloadPath);
+        cachedToolPath = await tools.cacheDir(terraformUnzippedPath, terraformToolName, version);
+    }
+
+    let terraformPath = findTerraformExecutable(cachedToolPath);
+    if (!terraformPath) {
+        throw new Error(tasks.loc("TerraformNotFoundInFolder", cachedToolPath));
+    }
+
+    fs.chmodSync(terraformPath, "777");
```

Comment:
Is this relevant in Windows?

Reply:
No, for Windows this was not required. I have added a check that will add the permissions only if it not a Windows agent.